applications: asset_tracker_v2: Add documentation for the UI module (#6444)
nrfconnect/sdk-nrf: applications/asset_tracker_v2/doc/ui_module.rst

CONFIG_UI_THREAD_STACK_SIZE - UI module thread stack size
   This option configures the UI module's internal thread stack size.

Module states
*************

The UI module continuously updates its internal state based on the mode and operating state of the application.
This ensures that the correct LED pattern is displayed when the application changes its state of operation.
The UI module has an internal state machine with the following states:

* ``STATE_ACTIVE`` - The application is in the active mode, revert to the active mode LED pattern.
* ``STATE_PASSIVE`` - The application is in the passive mode, revert to the passive mode LED pattern.

   * ``SUB_STATE_GNSS_ACTIVE`` - The application is performing a GNSS search, revert to the GNSS search LED pattern.
   * ``SUB_STATE_GNSS_INACTIVE`` - A GNSS search is not being performed.

* ``STATE_SHUTDOWN`` - The module has been shut down after receiving a request from the utility module.

Module events
*************

The :file:`asset_tracker_v2/src/events/ui_module_event.h` header file contains a list of various events sent by the module.

Dependencies
************

This module uses the following |NCS| libraries and drivers:

> * :ref:`:ref:`dk_buttons_and_leds_readme`

Reviewer: ```suggestion
* :ref:`dk_buttons_and_leds_readme`
```
Yet another whoops. I missed the extra `:ref:` and of course the doc build failed because of it.

.. _asset_tracker_v2_ui_module:

User Interface module
#####################

.. contents::
   :local:
   :depth: 2

> The User Interface module controls and monitors UI elements on the nRF9160 DK and Thingy:91.

Reviewer: ```suggestion
The User Interface module controls and monitors the UI elements on the nRF9160 DK and Thingy:91.
```

Features
********

This section documents the various features implemented by the module.

Button presses
==============

> Whenever a button is pressed on a supported board, the UI module notifies the rest of the application with the :c:enum:`UI_EVT_BUTTON_DATA_READY` event that contains
> information about the button press.
> The buttons that are utilized by the module and their related functionality are listed in the following table:

Reviewer: ```suggestion
When a button is pressed on a supported board, the UI module notifies the rest of the application with the :c:enum:`UI_EVT_BUTTON_DATA_READY` event that contains information about the button press.
The buttons used by the module and their functionality are listed in the following table:
```

> .. _button_behavior::

Reviewer: ```suggestion
.. _button_behavior:
```

+========+=====================================+==================================================================================================================+
| 1      | Send a message to the cloud service | Send message to the cloud service.                                                                               |
+--------+-------------------------------------+------------------------------------------------------------------------------------------------------------------+
| 2      |                                     | Send message to the cloud service.                                                                               |
|        |                                     +------------------------------------------------------------------------------------------------------------------+
|        |                                     | Fake movement. For testing purposes, the nRF9160 DK does not have an external accelerometer to trigger movement. |
+--------+-------------------------------------+------------------------------------------------------------------------------------------------------------------+

> .. _led_indication::

Reviewer: ```suggestion
.. _led_indication:
```

> LED Indication

Reviewer: ```suggestion
LED indication
```

==============

> The module supports multiple LED patterns to visualize the operating state of the application, listed in the table below:

Reviewer: ```suggestion
The module supports multiple LED patterns to visualize the operating state of the application.
```

> The following table describes the LED states supported by the module:

Reviewer: ```suggestion
The following table describes the supported LED states:
```

+---------------------------+-------------------------+-----------------------+
| State                     | Thingy:91 RGB LED       | nRF9160 DK solid LEDs |
+===========================+=========================+=======================+
| LTE connection search     | Yellow, blinking        | LED1 blinking         |
+---------------------------+-------------------------+-----------------------+
| GNSS fix search           | Purple, blinking        | LED2 blinking         |
+---------------------------+-------------------------+-----------------------+
| Publishing data           | Green, blinking         | LED3 blinking         |
+---------------------------+-------------------------+-----------------------+
| Active mode               | Light blue, blinking    | LED4 blinking         |
+---------------------------+-------------------------+-----------------------+
| Passive mode              | Dark blue, slow blinking| LED4 slow blinking    |
+---------------------------+-------------------------+-----------------------+

> | Error                     | Red, on                 | all 4 LEDs blinking   |
> +---------------------------+-------------------------+-----------------------+
> | Completion of FOTA Update | White, rapid blinking   | all 4 LEDs on         |

Reviewer 2: ```suggestion
| Error                     | Red, static             | all 4 LEDs blinking   |
+---------------------------+-------------------------+-----------------------+
| Completion of FOTA Update | White, rapid blinking   | all 4 LEDs on         |
```

> | Completion of FOTA Update | White, rapid blinking   | all 4 LEDs on         |

Reviewer: ```suggestion
| Completion of FOTA update | White, rapid blinking   | all 4 LEDs on         |
```

+---------------------------+-------------------------+-----------------------+

Configuration options
*********************

.. _CONFIG_UI_THREAD_STACK_SIZE:

CONFIG_UI_THREAD_STACK_SIZE - UI module thread stack size
   This option configures the UI module's internal thread stack size.

Module states
*************

The UI module continuously updates its internal state based on the mode and operating state of the application.

> This is to ensure that the correct LED pattern is displayed and reverted to when the application changes its state of operation.

Reviewer: ```suggestion
This ensures that the correct LED pattern is displayed when the application changes its state of operation.
```

The UI module has an internal state machine with the following states:

* ``STATE_ACTIVE`` - The application is in the active mode, revert to the active mode LED pattern.
* ``STATE_PASSIVE`` - The application is in the passive mode, revert to the passive mode LED pattern.

   * ``SUB_STATE_GPS_ACTIVE`` - The application is performing a GPS search, revert to the GPS search LED pattern.
   * ``SUB_STATE_GPS_INACTIVE`` - A GPS search is not being performed.

> * ``STATE_SHUTDOWN`` - The module has been shut down after receiving a request to do so from the utility module.

Reviewer: ```suggestion
* ``STATE_SHUTDOWN`` - The module has been shut down after receiving a request from the utility module.
```
